core/src/main/java/com/salesforce/slds/validation/validators/impl/recommendation/MobileFriendlyValidator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.slds.validation.validators.impl.recommendation;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.salesforce.slds.shared.models.context.Context;
+import com.salesforce.slds.shared.models.core.Bundle;
+import com.salesforce.slds.shared.models.core.Entry;
+import com.salesforce.slds.shared.models.core.HTMLElement;
+import com.salesforce.slds.shared.models.core.Input;
+import com.salesforce.slds.shared.models.recommendation.Action;
+import com.salesforce.slds.shared.models.recommendation.ActionType;
+import com.salesforce.slds.shared.models.recommendation.Item;
+import com.salesforce.slds.shared.models.recommendation.Recommendation;
+import com.salesforce.slds.shared.utils.ResourceUtilities;
+import com.salesforce.slds.validation.validators.interfaces.RecommendationValidator;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Component
+@Lazy
+public class MobileFriendlyValidator implements RecommendationValidator, InitializingBean {
+    Map<String, List<String>> componentsRegistry;
+
+    @Override
+    public List<Recommendation> matches(Entry entry, Bundle bundle, Context context) {
+        if (entry.getEntityType() != Entry.EntityType.LWC) {
+            return Lists.newArrayList();
+        }
+
+        return entry.getInputs().stream().filter(input -> input.asElement() != null)
+                .map(Input::asElement)
+                .filter(htmlElement -> !isMobileFriendly(htmlElement))
+                .map(htmlElement -> {
+                    Recommendation.RecommendationBuilder builder = Recommendation.builder()
+                            .input(htmlElement);
+
+                    String tag = htmlElement.getContent().tagName();
+
+                    Action action = Action.builder()
+                            .value(htmlElement.getContent().outerHtml()).range(htmlElement.getRange())
+                            .description(tag
+                                    + " is known to have issues on mobile devices. Consider using a replacement " +
+                                    "or create a custom component to use instead.")
```

Comment:
Is it ok to hardcode this string here? Do we need to handle non-English locales?

Reply:
@billyma and I had a chance to talk about this before. For the server side on LSP we need to pass a token from the client in order to determine which locale to use. That facility is not implemented yet. When it becomes available, and we can consider contribute this feature, I think then we can externalize messages into a string resource file.